src/components/tools/geometry-tool/geometry-toolbar.tsx

```diff
@@ -0,0 +1,124 @@
+  deleteDisabled: boolean;
+  onDuplicateClick: () => void;
+  duplicateDisabled: boolean;
+}
+interface IState {
+  showSettings: boolean;
+}
+
+@inject("stores")
+@observer
+export class GeometryToolbarView extends BaseComponent<IProps, IState> {
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      showSettings: false,
+    };
+  }
+  public render() {
+    return (
+      <div className="geometry-toolbar" data-test="geometry-toolbar" onMouseDown={this.handleMouseDown}>
+        <div className="toolbar-buttons">
+          {this.renderToolHeader(false)}
+          {this.renderToolButton("Select", "selection", null,
+                                  false, false, true)}
+          {this.renderToolButton("Point", "point", null,
+                                  false, false, true)}
+          {this.renderToolButton("Polygon", "polygon", null,
+                                  false, false, true)}
+          {this.renderToolButton("Duplicate", "duplicate", this.props.onDuplicateClick,
+                                  false, this.props.duplicateDisabled, false)}
+          {this.renderToolButton("Line Label", "line-label", null,
+                                  false, false, true)}
+          {this.renderToolButton("Angle Label", "angle-label", this.props.onAngleLabelClick,
+                                  this.props.angleLabelSelected, this.props.angleLabelDisabled, false)}
+          {this.renderToolButton("Movable Line", "movable-line", null,
+                                  false, false, true)}
+          {this.renderToolButton("Draw", "draw", null,
+                                  false, false, true)}
+          {this.renderToolButton("Text", "text", null,
+                                  false, false, true)}
+          {this.renderToolButton("Delete", "delete", this.props.onDeleteClick,
+                                  false, this.props.deleteDisabled, false)}
+        </div>
+        { this.state.showSettings
+          ? <div className="settings">
+              <div className="settings-header">
+                Settings
+              </div>
+            </div>
+          : null
+        }
+      </div>
+    );
+  }
+
+  private renderToolHeader = (showSettings: boolean) => {
+    return (
+      <div className="toolbar-header">
+        <div className="graph-tile">
+          <svg className="header-icon">
+            <use xlinkHref="#icon-graph-tile"/>
+          </svg>
+        </div>
+        {showSettings
+          ? <div className="settings-button" title="Settings"
+                 onClick={this.handleSettingsButton}>
+              <svg className="icon">
+                <use xlinkHref="#icon-settings"/>
+              </svg>
+            </div>
+          : null
+        }
+      </div>
+    );
+  }
+
+  private renderToolButton = (toolName: string, toolClass: string, handleClick: any,
+                              selected: boolean, disabled: boolean, hidden: boolean) => {
```

Comment:
As mentioned above, I'd be inclined to define the function something like this:
```typescript
interface IRenderToolButtonParams {
  onClick: (e: MouseEvent) => void;
  selected: boolean;
  disabled: boolean;
  hidden: boolean;
}

private renderToolButton = (toolName: string, toolClass: string, params: IRenderToolButtonParams) => {
}
```

Reply:
👍